add automated tests for presearch and algorand

A-Harby opened the pull request:
### Description

Add the automated tests for algorand and presearch application, (following same code struture as https://github.com/threefoldtech/tfgrid-sdk-ts/pull/2890).

The tests are different in that they ssh to the deployed VM and assert that the application is running with the correct version and network.

zaelgohary reviewed (changes requested):
I tried running Algorand test and it failed w this err. Is this normal? Same with Presearch.

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/40770501/9aa063c0-08c1-4ffd-a383-45294948b4a3)

A-Harby commented:
> I tried running Algorand test and it failed w this err. Is this normal? Same with Presearch.
> 
> ![image](https://private-user-images.githubusercontent.com/40770501/343700838-9aa063c0-08c1-4ffd-a383-45294948b4a3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk0OTc4NTQsIm5iZiI6MTcxOTQ5NzU1NCwicGF0aCI6Ii80MDc3MDUwMS8zNDM3MDA4MzgtOWFhMDYzYzAtMDhjMS00ZmZkLWEzODMtNDUyOTQ5NDhiNGEzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI3VDE0MTIzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdmNWI2ZTM1MDc1YjViMDMwZGZlMjUxMDdhNmQ4MzIwNTRmZTY1ZDAzNDc3NWUwMGUyNzc1MDgyZjhiYTE4ZmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.dR6rU73hC9mZD8pD-mqXtY5jP92oLrC-ek6xq8jUoXk)

Did you use the correct SSH key type? you should use the ssh-ed25519 not the normal one.

zaelgohary reviewed (commented):
The test would fail if SSH key needs permission.

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/40770501/eb4bc574-6259-4164-ab34-8c8ce3df94d8)

0oM4R reviewed (commented):
![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/d09fc4ef-4808-4816-8e19-13ccd74e6a8a)

this [package](https://www.npmjs.com/package/semver#prerelease-identifiers) may help

0oM4R reviewed (commented):
and please notice that the current logic overwrite the provided config if one of the `mnemonic`, `network`, or `ssh_key` are not provided in env.

we may change that logic to get the missing `only` form config.json 

https://github.com/threefoldtech/tfgrid-sdk-ts/blob/828f0a0efacfd8a27e58a5befcd931782d2885f7/packages/grid_client/tests/client_loader.ts#L15-L26

0oM4R commented on packages/grid_client/tests/config.json:
remove those config please

0oM4R commented on packages/grid_client/tests/client_loader.ts:
```diff
@@ -11,17 +11,18 @@ const mnemonic = env.MNEMONIC;
 let config;
+const defaultConfig = JSON.parse(fs.readFileSync(path.join(__dirname, "./config.json"), "utf-8"));
 
 if (!network || !mnemonic || !ssh_key) {
```
this won't fix the issue, 
![Screenshot from 2024-07-04 08-10-28](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/387e085f-eec5-4747-b3ec-3c9a6e8783df)

i think its better to revert this changes and mention on docs that all credential must be in one place

0oM4R reviewed (commented):
![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/55fbd5f0-0b08-4579-9a48-4f70c3a49f90)

presearch failed

0oM4R commented:
can you please validate if the machine that runs these tests has ygg installed and configured, 
I forgot to install ygg and got some connection errors, It was unclear if it came from the deployment or from my side

A-Harby commented:
> can you please validate if the machine that runs these tests has ygg installed and configured, I forgot to install ygg and got some connection errors, It was unclear if it came from the deployment or from my side

As we discussed, I'm going to mention that on the Readme to make sure that ygg is needed to be installed with all of the other needed packages.

Also, I added a few tries in the trim of the Docker hash: https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3030/commits/a3b4659951b9855546386019f4b30059e6116f8b.

And I have added the required notes here: https://github.com/threefoldtech/tfgrid-sdk-ts/pull/2890/commits/e8e35cba245a7c5f34c41ced7ce9cd733a689767.

zaelgohary reviewed (approved):
Good job! @A-Harby
